gensim/models/keyedvectors.py

```diff
@@ -1130,16 +1274,24 @@ def evaluate_word_analogies(self, analogies, restrict_vocab=300000, case_insensi
                 self.vocab = ok_vocab
                 ignore = {a, b, c}  # input words to be ignored
                 predicted = None
-                # find the most likely prediction using 3CosAdd (vector offset) method
-                # TODO: implement 3CosMul and set-based methods for solving analogies
-                sims = self.most_similar(positive=[b, c], negative=[a], topn=5, restrict_vocab=restrict_vocab)
+                # find the most likely prediction using 3CosAdd (vector offset) method or 3CosMul method
+
+                if method == '3CosAdd':
+                    sims = self.most_similar(positive=[b, c], negative=[a], topn=topk, restrict_vocab=restrict_vocab)
+                elif method == '3CosMul':
+                    sims = self.most_similar_cosmul(positive=[b, c], negative=[a], topn=topk,
```

Comment:
Hanging indent (here and elsewhere): if the line is too long, start the parameters on a new line. See [PEP8 examples of hanging indent](https://www.python.org/dev/peps/pep-0008/#indentation).